test/integration/master/kms_transformation_test.go

```diff
@@ -206,9 +206,7 @@ resources:
 		t.Fatalf("failed to create mock of KMS Plugin #1: %v", err)
 	}
 
-	if err := pluginMock1.Start(); err != nil {
-		t.Fatalf("Failed to start kms-plugin, err: %v", err)
-	}
+	go pluginMock1.Start()
```

Comment:
How much time does this save?

I think this will come down to the trade-off between shaving-off some time vs. lose in the error reporting clarity.

Reply:
Very little, like 1-2s. It couldn't bring the time down under 30s(originally 40-45s). I can close this PR if we need to modify strategy.

Reply:
I think the fundamental problem is that we hard-code the TTL for healthchecks:
```
kmsPluginHealthzNegativeTTL  = 3 * time.Second
kmsPluginHealthzPositiveTTL  = 20 * time.Second
```
I had a PR that proposed exposing these values in the EncryptionConfig, but the general opinion was that this was not something we wanted to expose to end-users.
Having these values available in the test setup would probably address this issue of slowness.

Reply:
This test will take at least 20s to run due to the hard-coded value for `kmsPluginHealthzPositiveTTL = 20 * times.Second`.
In other words, to simulate a failure of the kms-plugin (go from healthy to unhealthy), the test would need to wait 20s to confirm the transition.

@liggitt and @enj Should we take another look at #86368 which would address this issue.

Reply:
I'd start by making a test-only way to shorten that rather than jump right to config

Reply:
Sure, we could try the following approach - going from unhealthy to health - this could save 17s since the negative TTL is just 3 seconds. @ZhiFeng1993 does this make sense?

Reply:
hmm @immutableT  In this test I don't think either `kmsPluginHealthzNegativeTTL` or `kmsPluginHealthzNegativeTTL` is being used. It's using `mustBeHealthy` and `mustBeUnhealthy` that are defined in this file. See here:
https://github.com/kubernetes/kubernetes/blob/master/test/integration/master/kms_transformation_test.go#L261

And in both cases the waiting time is `wait.ForeverTestTimeout`, which is 30s, We can change that value to shorten the waiting time. Do I miss anything?

Reply:
I lowered `wait.ForeverTestTimeout` to 20s but haven't seen obvious effect. That's only for waiting time.

Reply:
@immutableT  I think you are looking at the wrong test- `TestKMSPluginHealthzTTL`. Or am I wrong?

Reply:
I am looking at `TestKMSPluginHealthzTTL` - this is the slow test, right?

You are correct that the test itself does not use the values of `kmsPluginHealthzPositiveTTL` and `kmsPluginHealthzNegativeTTL`, rather the logic under test is using these hard-coded values to determine when to re-check the health of the plugin. 

Essentially, this tests goes through the following phases:
1. Setup (fast)
2. Assert that OK is returned for a healthy plugin (fast)
3. Place a plugin into failed state (fast)
4. Validate that !OK is returned for a failed plugin (slow, because we kube-apiserver will not re-check the status until the PositiveTTL expires - this is what the wait.ForeverTestTimeout does).

As I mentioned above, ideally we should expose these TTL values in the config so that users and tests could adjust this.
However, there the current opinion is that we should not complicate the API  of the plugin with this level of details.
So, we could try the following approach:

1. Setup with kms-plugin in the failed state (fast)
2. Assert that !OK is returned for a failed plugin (faster than before 2s vs. 20s)
3. Place a plugin into the healthy state (fast)
4. Validate that OK is returned for a failed plugin (faster than before 2s vs. 20s).

Reply:
Thanks @immutableT  I think we are looking at: `k8s.io/kubernetes/test/integration/master.TestKMSHealthz`, not `TestKMSPluginHealthzTTL`

Reply:
`TestKMSPluginHealthzTTL` is the one in `config_test.go`